### PR TITLE
Opt gluniform array upload

### PR DESCRIPTION
**Pull request by juj**
Extend the array forms of uniform uploads to use the temp view buffer for larger array sizes, and unroll a bit. Also fix a signature typo.

Ran browser suite to pass.

### CODE REVIEW

**Comment by kripken** [comment]
`miniTempBuffer`s are used in various places that seem to depend on their size being 16. Is your change here to that size safe?

File: src/library_gl.js [hunk 1]
```diff
@@ -2761,11 +2780,26 @@ var LibraryGL = {
...
     var view;
-    if (count === 1) {
-      // avoid allocation for the common case of uploading one uniform matrix
-      view = GL.miniTempBufferViews[15];
-      for (var i = 0; i < 16; i++) {
-        view[i] = {{{ makeGetValue('value', 'i*4', 'float') }}};
+    if (16*count <= GL.MINI_TEMP_BUFFER_SIZE) {
+      // avoid allocation when uploading few enough uniforms
+      view = GL.miniTempBufferViews[16*count-1];
+      for(var i = 0; i < 16*count; i += 16) {
```
Comment (kripken) [inline src/library_gl.js]: space between `for` and `(`

File: src/library_gl.js [hunk 2]
```diff
@@ -2761,11 +2780,26 @@ var LibraryGL = {
...
+        view[i] = {{{ makeGetValue('value', '4*i', 'float') }}};
+        view[i+1] = {{{ makeGetValue('value', '4*i+4', 'float') }}};
+        view[i+2] = {{{ makeGetValue('value', '4*i+8', 'float') }}};
+        view[i+3] = {{{ makeGetValue('value', '4*i+12', 'float') }}};
```
Comment (kripken) [inline src/library_gl.js]: this does 16 loads and stores vs one `.subarray`. Did you benchmark it?

**Comment by kripken** [comment]
lgtm aside from that question on speed for the largest of these new loops.

**Comment by juj** [comment]
Did benchmarking in the glgears sample, by adapting that to utilize an array of 16 matrices, which is the upper limit of the new code path `GL.MINI_TEMP_BUFFER_SIZE` (16*16=256). Uploading an array of 16 matrices (x1000 times per frame for more stable measures) takes 5.5msecs on average with the new path, and 15.5msecs with the old.

Uploading one matrix only takes 2.2msecs, which suggests an increase of 0.8msecs per extra matrix. Overall this suggests that the cutoff limit for `GL.MINI_TEMP_BUFFER_SIZE` could probably be vastly larger than now, but the value of 256 is already enough to replace all array uploads in UE4 engine with the fast path and avoid .subarray() anywhere.

Btw, this is interestingly similar to #4127 (ignoring the simd portion there), and in our memcpy, the cutoff has been at 4k bytes ballpark, hinting that `GL.MINI_TEMP_BUFFER_SIZE` could even be bumped as far high as 1024, but I'm going to be conservative here for now and keep it at 256 based on the UE4 and glgears cases.

File: src/library_gl.js (hunk 1 above)
Comment (juj) [inline src/library_gl.js]: Check.

**Pull request merged**